monkey/infection_monkey/master/control_channel.py

```diff
@@ -36,7 +36,8 @@ def should_agent_stop(self) -> bool:
         if not self._control_channel_server:
             logger.error("Agent should stop because it can't connect to the C&C server.")
             return True
-        return self._island_api_client.get_agent_signals(self._agent_id) is not None
+        agent_signals = self._island_api_client.get_agent_signals(self._agent_id)
+        return agent_signals.terminate is not None
```

Comment:
For now the agent doesn't use it. We may in the future. For example, we can set a hard limit on how long we're willing to wait for the agent to shut down. The agent can compare current time to the signal time and decide whether or not to kill everything.

The other thing is that we don't want the Island to assume how the data will be used. The Island needs to know when a signal was sent so it makes that data available. We don't want the island to send a boolean because that's all the agent needs. The island should just make the data available.